migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs

```diff
@@ -33,15 +33,19 @@ impl Connection {
         &mut self,
         circumstances: BitFlags<super::Circumstances>,
         params: &super::Params,
+        namespaces: Option<Namespaces>,
     ) -> ConnectorResult<SqlSchema> {
         use sql_schema_describer::{postgres as describer, DescriberErrorKind, SqlSchemaDescriberBackend};
         let mut describer_circumstances: BitFlags<describer::Circumstances> = Default::default();
         if circumstances.contains(super::Circumstances::IsCockroachDb) {
             describer_circumstances |= describer::Circumstances::Cockroach;
         }
 
+        let namespaces_vec = Namespaces::to_vec(namespaces, String::from(params.url.schema()));
+        let namespaces_str: Vec<&str> = namespaces_vec.iter().map(AsRef::as_ref).collect();
```

Comment:
There must be a better way than this :sweat_smile:

Reply:
awkwardly, there isn't :/ The go-to way would be handing down a `Vec<&str>` from the top (so having `&str` instead of owned strings in `Namespaces`). That should be fine, but it's more diff so let's keep things this way for now. It's not a performance problem unless we decide it is after measuring.

Reply:
The reason why you can't interpret a `&[String]` as a `&[&str]` is that `String` and `&str` have different sizes in memory (`&str` is a pointer and a length, so `usize` * 2, whereas `String` has a pointer, a size and a capacity, so `usize` * 3)